### PR TITLE
Support WeCom Robot receiver in alertmanagerconfig CRD

### PR DESCRIPTION
**Pull request by magiceses**
## Description

The purpose of this pr is to enable alertmanager to send notifications to [WeCom Chatbot](https://developer.work.weixin.qq.com/document/path/91770#%E5%A6%82%E4%BD%95%E4%BD%BF%E7%94%A8%E7%BE%A4%E6%9C%BA%E5%99%A8%E4%BA%BA). For this purpose, i add a new Notifier and some new configs.

## Type of change

_What type of changes does your code introduce to the Prometheus operator? Put an `x` in the box that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [x] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [ ] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Changelog entry

_Please put a one-line changelog entry below. This will be copied to the changelog file during the release process._

```release-note
Support WeCom Robot receiver in alertmanagerconfig CRD
```

### CODE REVIEW

File: go.mod
Comment (hoperays) [inline go.mod:1]: Please do not modify go.mod in this patch, upgrade your local golang version if necessary.

File: pkg/alertmanager/amcfg.go [hunk 1]
```diff
@@ -639,19 +639,32 @@ func (cb *configBuilder) convertReceiver(ctx context.Context, in *monitoringv1al
...
+		DiscordConfigs:    discordConfigs,
+		SlackConfigs:      slackConfigs,
+		WebhookConfigs:    webhookConfigs,
+		WeChatConfigs:     weChatConfigs,
+		EmailConfigs:      emailConfigs,
+		VictorOpsConfigs:  victorOpsConfigs,
+		PushoverConfigs:   pushoverConfigs,
+		SNSConfigs:        snsConfigs,
+		TelegramConfigs:   telegramConfigs,
+		WeComRobotConfigs: weComRobotConfigs,
```
Comment (hoperays) [inline pkg/alertmanager/amcfg.go:668]: Leave a blank line to facilitate version upgrade or merge with other patches.

File: pkg/alertmanager/types.go [hunk 2]
```diff
@@ -96,19 +96,20 @@ type inhibitRule struct {
...
+	WebhookConfigs    []*webhookConfig    `yaml:"webhook_configs,omitempty" json:"webhook_configs,omitempty"`
+	WeChatConfigs     []*weChatConfig     `yaml:"wechat_configs,omitempty" json:"wechat_config,omitempty"`
+	EmailConfigs      []*emailConfig      `yaml:"email_configs,omitempty" json:"email_configs,omitempty"`
+	PushoverConfigs   []*pushoverConfig   `yaml:"pushover_configs,omitempty" json:"pushover_configs,omitempty"`
+	VictorOpsConfigs  []*victorOpsConfig  `yaml:"victorops_configs,omitempty" json:"victorops_configs,omitempty"`
+	SNSConfigs        []*snsConfig        `yaml:"sns_configs,omitempty" json:"sns_configs,omitempty"`
+	TelegramConfigs   []*telegramConfig   `yaml:"telegram_configs,omitempty" json:"telegram_configs,omitempty"`
+	DiscordConfigs    []*discordConfig    `yaml:"discord_configs,omitempty"`
+	WebexConfigs      []*webexConfig      `yaml:"webex_configs,omitempty"`
+	WeComRobotConfigs []*weComRobotConfig `yaml:"wecomrobot_configs,omitempty"`
```
Comment (hoperays) [inline pkg/alertmanager/types.go:113]: ditto.

File: pkg/alertmanager/types.go [hunk 3]
```diff
@@ -343,6 +344,14 @@ type telegramConfig struct {
 	HTTPConfig           *httpClientConfig `yaml:"http_config,omitempty" json:"http_config,omitempty"`
 }
 
+type weComRobotConfig struct {
+	VSendResolved  *bool             `yaml:"send_resolved,omitempty" json:"send_resolved,omitempty"`
+	HTTPConfig     *httpClientConfig `yaml:"http_config,omitempty" json:"http_config,omitempty"`
+	Message        string            `yaml:"message,omitempty" json:"message,omitempty"`
+	WebhookURL     string            `yaml:"webhook_url,omitempty" json:"webhook_url,omitempty"`
+	MaxMessageSize int               `yaml:"max_message_size,omitempty" json:"max_message_size,omitempty"`
```
Comment (hoperays) [inline pkg/alertmanager/types.go]: Field map for json is unuseful.

**Comment by hoperays** [comment]
LGTM

**Review by charlottexl — COMMENTED** [review]
LGTM

**Pull request merged**